### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by KUBAHO3**
**In this project I had:**

- Install Ruby on Rails framework.
- Understand Rails RESTful design and router.
- Use controllers to handle requests and render empty views.
- Use params from browser requests in a safe way.
- Use preprocessed html files with embedded Ruby code.
- Use layouts and templates for shared content.
- Use database migration files to maintain database schema.
- Use validations for models.
- Secure app from n+1 problems.
- Understand what ORM is.
- Write SQL queries with ActiveRecord.
- Set up associations between models.
- Build a web app that requires the user to log in.
- Use devise gem for authentication.

### CODE REVIEW

File: app/views/groups/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+          <p class="mt-14 text-xs"><%= group.created_at.strftime('%d %h %Y') %></p>
+        </div>
+
+        <p class="ml-16 grow-0 font-semibold">$<%= @total_amount[index] %> </p>
+      </div>
+      <% end %>
+    <% end %>
+  <% end %>
+
+  <%= link_to 'Add new category', new_group_path, class: "bg-green-400 p-2 w-96 text-center rounded text-slate-50 mt-6" %>
```
Comment (DammyShittu) [inline app/views/groups/index.html.erb]: - [x] Kindly ensure the new category button sticks to the bottom of the page as described in the design.

- [x] Kindly remove the sign-out button from the bottom of all your pages. Sign-out buttons do not belong there. They should be on the navbar.

  siss

File: app/views/entities/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    <div class="flex mt-4 justify-between">
+      <p>Payment</p>
+      <p class="text-slate-900 font-semibold">$<%= entity.amount %></p>
+  </div>
+  </div>
+  <hr class="mt-3">
+  <% end %>
+
+  <%= link_to 'Add new transaction', new_group_entity_path, class: "bg-green-400 w-96 text-center p-2 rounded text-slate-50 mt-6" %>
```
Comment (DammyShittu) [inline app/views/entities/index.html.erb]: - [x] Kindly ensure the add new transaction button sticks to the bottom of the page.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @KUBAHO3,

## Highlights
✔️ Splash screen
✔️ Passing tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by franclobo — APPROVED** [review]
Hi @KUBAHO3, 

# STATUS: APPROVED 😃 

leonardo-dicaprio-leo-dicaprio

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights

- Unit and integration tests passed ✔️
- Use PostgreSQL ✔️
- Use device for authentication and cancancan for authorization ✔️
- Good UI and UX ✔️


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**